src/app/nutrition/dialogs/food-log-rows.tsx

```diff
@@ -155,9 +182,17 @@ export function MealTypeRows({ summary }: { summary: MealTypeSummary }) {
   const checked = selectedFoodLogs.filter(
       (foodLog) => foodLog.loggedFood.mealTypeId == summary.id || summary.id == -1
   ).size > 0;
-  const intersect = summary.foods.filter((foodLog) => selectedFoodLogs.has(foodLog));
-  const indeterminate = checked && intersect.length != summary.foods.length;
-  const label = checked ? "Select none" : "Select all";
+  const indeterminate = checked && !summary.foods.every((foodLog) => selectedFoodLogs.has(foodLog));
+  const title = !checked || indeterminate ? "Select all" : "Select none";
+
+  async function copyTextToClipboard(e: MouseEvent) {
```

Comment:
Just a note that React (and Javascript event handlers in general) don't do anything with the async promise returned by the function. It'll get ignored, which doesn't matter in this case but for more complex actions we'd want to make sure any errors are reported and so on. OK to keep this as is, but I'll probably rewrite this (and a few other places) later to something like:

```ts
const copyTextToClipboard = reportAsyncErrors(async () {
 if ('clipboard' in navigator) {
  const button = e.target as HTMLButtonElement;

  if (button.textContent) {
    try {
      return await navigator.clipboard.writeText(button.textContent || "");
    } catch (e) {
      // copying blocked by browser due to security restrictions
      toast.error("Error copying to clipboard");
    }
  }
});
```

`reportAsyncErrors` doesn't exist yet, but it would add a `.catch()` in case your function didn't handle some error. This would allow us to enable a linter rule later on that warns you if you forget to await/then/catch a promise.